Dockerfile-rtla

```diff
@@ -1,13 +1,74 @@
-# Using fedora to get a recent version of the rtla package
-FROM fedora:39
+ARG BASE_IMAGE=fedora
+ARG RHEL_HOST=false
+ARG RTLA_UPSTREAM=false
+
+FROM ${BASE_IMAGE}
+
+ARG BASE_IMAGE
+ARG RHEL_HOST
+ARG RTLA_UPSTREAM
+RUN echo "Debug: BASE_IMAGE=${BASE_IMAGE}, RHEL_HOST=${RHEL_HOST}, RTLA_UPSTREAM=${RTLA_UPSTREAM}"
+
+# Use shell to check image_name conditions
+RUN if [[ ("$BASE_IMAGE" != "ubi9" && "$BASE_IMAGE" != "centos:stream9" && "$BASE_IMAGE" != "centos:stream9" && "$BASE_IMAGE" != "fedora") ]]; then \
+    echo "ERROR: Invalid BASE_IMAGE given: ${BASE_IMAGE}."; \ 
+    echo "ERROR: The build-arg BASE_IMAGE must be either 'ubi9' for registered host or 'centos:stream9' for non-rhel host"; \
+    exit 1; \
+    fi
+
+# Use shell to check rhel host build conditions
+RUN if [[ "$RHEL_HOST" == "false" && "$BASE_IMAGE" == "ubi9" ]]; then \
```

Comment:
I agree. When I wrote it I though it was clever. But, maybe its a little too complicated.

The main reason for this is to have a place to pull the current RHEL version of the tooling. If I removed the bits about RHEL_HOST and ubi9, and left this only as Fedora or cs9, would that be less complicated?

Reply:
Why not just fedora:39 as it was before?

Reply:
I can remove this. The only reason I was thinking we may want to use cs9 is to have a place to retrieve the current "stable" version of rtla.

Reply:
Another thing I just realized. rtla in fedora is probably very close to the upstream version? So maybe instead of building it from upstream, we can use the version in fedora when we need something closer to upstream?